plugins/out_parseable/parseable.c

```diff
@@ -0,0 +1,378 @@
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_utils.h>
+#include <fluent-bit/flb_slist.h>
+#include <fluent-bit/flb_time.h>
+#include <fluent-bit/flb_pack.h>
+#include <fluent-bit/flb_config_map.h>
+#include <fluent-bit/flb_metrics.h>
+#include <fluent-bit/flb_log_event_decoder.h>
+#include <fluent-bit/flb_event.h>
+#include <fluent-bit/flb_record_accessor.h>
+
+
+#include "parseable.h"
+
+static int cb_parseable_init(struct flb_output_instance *ins,
+                             struct flb_config *config, void *data)
+{
+    int ret;
+    struct flb_out_parseable *ctx = NULL;
+    (void) ins;
+    (void) config;
+    (void) data;
+
+    ctx = flb_calloc(1, sizeof(struct flb_out_parseable));
+    if (!ctx) {
+        flb_errno();
+        return -1;
+    }
+    ctx->ins = ins;
+
+    /* Read in config values */
+    ret = flb_output_config_map_set(ins, (void *) ctx);
+    if (ret == -1) {
+        flb_free(ctx);
+        return -1;
+    }
+
+    flb_plg_info(ctx->ins, "Configured port: %d", ctx->server_port);
+
+    ctx->upstream = flb_upstream_create(config,
+                                        ctx->server_host,
+                                        ctx->server_port,
+                                        FLB_IO_TCP | FLB_IO_ASYNC,
+                                        NULL);
+
+    ctx->upstream->base.net.connect_timeout = ctx->connect_timeout;
+    ctx->upstream->base.net.accept_timeout = ctx->accept_timeout;
+
+    flb_plg_info(ctx->ins, "Timeouts - Connect: %ds, Accept: %ds", ctx->connect_timeout, ctx->accept_timeout);
+
+    /* Set retry limit */
+    char retry_limit_str[16];
+    snprintf(retry_limit_str, sizeof(retry_limit_str), "%d", ctx->retry_limit);
+    flb_output_set_property(ins, "Retry_Limit", retry_limit_str);
+    
+    flb_plg_info(ctx->ins, "Retry limit set to: %d", ctx->retry_limit);
+
+    if (!ctx->upstream) {
```

Comment:
You need to move this conditional to be right after the call to `flb_upstream_create`.